### PR TITLE
Add unsafe_hash alias for class-wide hash

### LINKED ISSUE #1003: Add unsafe_hash for PEP 681
I've been asked to add `unsafe_hash` as an alias for `hash` **at class level** for PEP 618 (data class transforms).

I'm not enthusiastic about the lack of symmetry between class-level and field-level hash but this is not the hill I'm dying on.

### PR DESCRIPTION
**Pull request by hynek**
# Summary

`attr.s(unsafe_hash=True)` is now equivalent to `attr.s(hash=True)` to standardize based on [PEP 681](https://peps.python.org/pep-0681/).

Fixes #1003

# Pull Request Check List

- [x] Added **tests** for changed code.
  Our CI fails if coverage is not 100%.
- [x] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/main/tests/strategies.py).
- [x] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
    - [x] ...and used in the stub test file `tests/typing_example.py`.
    - [x] If they've been added to `attr/__init__.pyi`, they've *also* been re-imported in `attrs/__init__.pyi`.
- [x] Updated **documentation** for changed code.
    - [x] New functions/classes have to be added to `docs/api.rst` by hand.
    - [x] Changes to the signature of `@attr.s()` have to be added by hand too.
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
          Find the appropriate next version in our [``__init__.py``](https://github.com/python-attrs/attrs/blob/main/src/attr/__init__.py) file.
- [x] Documentation in `.rst` files is written using [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/main/changelog.d).
- [x] Consider granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so maintainers can fix minor issues themselves without pestering you.

### CODE REVIEW

**Comment by hynek** [comment]
@debonte does this cover what you asked me about in *checks notes* April? 🙈

File: docs/hashing.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+1. Two objects that are equal, **must** have the same hash.
+   This means that if `x == y`, it *must* follow that `hash(x) == hash(y)`.
+
+   By default, Python classes are compared *and* hashed by their `id`.
+   That means that every instance of a class has a different hash, no matter what attributes it carries.
+
+   It follows that the moment you (or *attrs*) change the way equality is handled by implementing `__eq__` which is based on attribute values, this constraint is broken.
+   For that reason Python 3 will make a class that has customized equality unhashable.
+   Python 2 on the other hand will happily let you shoot your foot off.
+   Unfortunately, *attrs* still mimics (otherwise unsupported) Python 2's behavior for backward compatibility reasons if you set `hash=False`.
```
Comment (debonte) [inline docs/hashing.md:35]: You mention the `hash` parameter here and line 58, but `unsafe_hash` in a couple other places? Maybe you should consistently use one or the other? Or maybe mention that they are equivalent?

**Comment by debonte** [comment]
> @debonte does this cover what you asked me about in _checks notes_ April? 🙈

:) I added one minor comment about the documentation. Otherwise, this looks good to me. Thanks for making this change!

File: docs/hashing.md (hunk 1 above)
Comment (hynek) [reply docs/hashing.md:35]: ugh right thanks; i thought i caught them all!

**Pull request merged**